joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -221,6 +246,37 @@ JointTrajectoryController::on_configure(const rclcpp_lifecycle::State & previous
   // TODO(karsten1987): no lifecyle for subscriber yet
   // joint_command_subscriber_->on_activate();
 
+  // State publisher
+  {
+    int state_publish_rate = 50;
+    if (lifecycle_node_->has_parameter("state_publish_rate")) {
+      state_publish_rate = lifecycle_node_->get_parameter("state_publish_rate").get_value<int>();
+      RCLCPP_INFO_STREAM(
+        logger, "Controller state will be published at " <<
+          state_publish_rate << "Hz.");
+    }
+    state_publisher_period_ = rclcpp::Duration(1.0 / static_cast<double>(state_publish_rate));
+
+    publisher_ = lifecycle_node_->create_publisher<JTrajControllerStateMsg>(
+      "state", rclcpp::SystemDefaultsQoS());
+    state_publisher_.reset(new StatePublisher(publisher_));
```

Comment:
Is there a reason for creating a copy of a local object only to shove it into a unique pointer?

How about this, instead of these two lines?

`state_publisher_ = std::make_unique(lifecycle_node_->create_publisher<ControllerStateMsg>("state", rclcpp::SystemDefaultsQoS()));`

Reply:
There's some confusion here I think?; `publisher_` is of type LifecyclePublisher, and `StatePublisher` is of type RealtimePublisher, so there's no local copy.